Flaky Test: TestRestartMonitor

The new test `TestRestartMonitor` seems to fail somewhat regularly during our 'Cgroups v2 & SELinux' integration run, which is the only run that occurs via nested virt (Fedora on macos via vagrant).  Not claiming that is the culprit, but one data point about what might be different.

```
2021-03-09T16:32:40.5021020Z     default:     restart_monitor_linux_test.go:88: 2021-03-09 16:32:39.852326838 +0000 UTC m=+98.660313953: status={"stopped" '\u0089' "2021-03-09 16:32:39.848159102 +0000 UTC"}
2021-03-09T16:32:41.5224630Z     default:     restart_monitor_linux_test.go:88: 2021-03-09 16:32:40.877046477 +0000 UTC m=+99.685033613: status={"stopped" '\u0089' "2021-03-09 16:32:39.848159102 +0000 UTC"}
2021-03-09T16:32:42.5810400Z     default:     restart_monitor_linux_test.go:88: 2021-03-09 16:32:41.936167085 +0000 UTC m=+100.744154195: status={"stopped" '\u0089' "2021-03-09 16:32:39.848159102 +0000 UTC"}
2021-03-09T16:32:43.5916370Z     default:     restart_monitor_linux_test.go:88: 2021-03-09 16:32:42.946341209 +0000 UTC m=+101.754328313: status={"stopped" '\u0089' "2021-03-09 16:32:39.848159102 +0000 UTC"}
2021-03-09T16:32:44.5943680Z     default:     restart_monitor_linux_test.go:88: 2021-03-09 16:32:43.949601583 +0000 UTC m=+102.757588692: status={"stopped" '\u0089' "2021-03-09 16:32:39.848159102 +0000 UTC"}
2021-03-09T16:32:45.6017510Z     default:     restart_monitor_linux_test.go:88: 2021-03-09 16:32:44.956206848 +0000 UTC m=+103.764193943: status={"stopped" '\u0089' "2021-03-09 16:32:39.848159102 +0000 UTC"}
2021-03-09T16:32:46.8142670Z     default:     restart_monitor_linux_test.go:88: 2021-03-09 16:32:46.169687386 +0000 UTC m=+104.977674523: status={"stopped" '\u0089' "2021-03-09 16:32:39.848159102 +0000 UTC"}
2021-03-09T16:32:47.8356810Z     default:     restart_monitor_linux_test.go:88: 2021-03-09 16:32:47.190661019 +0000 UTC m=+105.998648157: status={"stopped" '\u0089' "2021-03-09 16:32:39.848159102 +0000 UTC"}
2021-03-09T16:32:48.8389790Z     default:     restart_monitor_linux_test.go:86: task TestRestartMonitor not found: not found
2021-03-09T16:32:48.8677920Z     default: --- FAIL: TestRestartMonitor (10.25s)
```

// cc: @AkihiroSuda

--- a/restart_monitor_linux_test.go
+++ b/restart_monitor_linux_test.go
@@ -82,15 +82,19 @@ version = 2
 	deadline := begin.Add(interval).Add(epsilon)
 	for time.Now().Before(deadline) {
 		status, err := task.Status(ctx)
+		now := time.Now()
 		if err != nil {
-			t.Fatal(err)
-		}
-		t.Logf("%v: status=%q", time.Now(), status)
+			// ErrNotFound is expected here, because the restart monitor
+			// temporarily removes the task before restarting.
+			t.Logf("%v: err=%v", now, err)
+		} else {
+			t.Logf("%v: status=%q", now, status)
 
-		if status.Status == Running {
-			elapsed := time.Since(begin)
-			t.Logf("the task was restarted after %s", elapsed.String())
-			return
+			if status.Status == Running {
+				elapsed := time.Since(begin)
+				t.Logf("the task was restarted within %s", elapsed.String())
+				return
+			}
 		}
 		time.Sleep(epsilon)
 	}

